qiime2/plugin/plugin.py

```diff
@@ -135,6 +138,22 @@ def register_views(self, *views, citations=None):
             if is_format:
                 self.formats[name] = FormatRecord(format=view, plugin=self)
 
+    def register_validator(self, semantic_expression):
+        if not is_semantic_type(semantic_expression):
+            raise TypeError('%s is not a Semantic Type' % semantic_expression)
+
+        def decorator(validator):
+            for semantic_type in semantic_expression:
+                if semantic_type not in self.validators:
+                    self.validators[semantic_type] = []
+                self.validators[semantic_type].append(ValidatorRecord(
```

Comment:
I would probably construct a single ValidatorRecord, just so that their object IDs are shared between partitions. This isn't likely to be all that useful implementation wise, but I expect it would simplify testing and debug.

Reply:
(instead of making a new one in each iteration of the loop)